### DISCUSSION TITLE
we have a new folder (metadata) in our repo called GenAIPromptTemplates but sdg is not picking it up. Help? Thank you!

### DISCUSSION THREAD

**Discussion by sheidasf** [scolladon/sfdx-git-delta#905] [Q&A]
we have a new folder in our repo called GenAIPromptTemplates but sdg is not picking it up. It is not in the packcage.xml that is generated. Thanks community!

**Comment by scolladon** [comment]
Hi @sheidasf 

[GenAIPromptTemplate](https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_genaiprompttemplate.htm) are handled by SGD (since [v5.40.0](https://github.com/scolladon/sfdx-git-delta/releases/tag/v5.40.0)).

To get it you need to use the api version 60+ or above.
The plugin deduce the version from either the command line, then from the `sfdx-project.json` file and if none of those are set it takes the latest available minus preview version.

According to the [doc](https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_genaiprompttemplate.htm), GenAIPromptTemplate files should be in a `genAiPromptTemplates` folder with the suffix `. genAiPromptTemplate`. Could you make sure it is the case please ?

Could you show how you execute the plugin please ? (what parameters)
Could you show the content of the `sfdx-project.json` file please ?

**Comment by sheidasf** [comment]
Thank you! So much!

I have a lot of confusion on how the package.xml is generated. (albeit I am not the expert in git from a commit perspective), but I don’t know what it looks at to generate the package.xml. if you have a dumb down version of this material for not-so-smart people like me, I would love to see it.

For example, this time around it didn’t pick up the files from that folder, and I am looking at my options on how to deploy them, since they were already committed and merged to the branch.

ANY WAY!!!

This is the line from our yaml file:
```yaml
          - mkdir changed-sources && sfdx sgd:source:delta -t HEAD  -f $LAST_DEPLOYED_COMMIT -r . --output changed-sources/ -d --loglevel=trace
```

Our api was indeed set to 59.
```json
{
  "packageDirectories": [
    {
      "path": "force-app",
      "default": true
    }
  ],
  "namespace": "",
  "sfdcLoginUrl": "https://test.salesforce.com",
  "sourceApiVersion": "59.0"
}
```



Best,
-Sheida

**Reply by scolladon** [reply]
Ok so I think the plugin does not treat [GenAIPromptTemplate](https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_genaiprompttemplate.htm) metadata because of the API version used.

Could either set the api version (`--api-version` parameter) or change the `sourceApiVersion` attribut in the `sfdx-project.json` file to version `61.0` and retry please ?

**Comment by sheidasf** [comment]
Hi Sebastian,

Thank you so much for your quick response. I updated the package to 61 and it indeed picked it up, but may be salesforce is not quite ready for sfdx deploy of the prompts. But no, I get an error:

Error (1): /opt/atlassian/pipelines/agent/build/changed-sources/force-app/main/default/genAIPromptTemplates/Account_Open_Opportunities_Slalom_Sales.genAiPromptTemplate-meta.xml: Could not infer a metadata type
612

On this page, https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_genaiprompttemplate.htm  under special access rules, it says ‘To send GenAiPromptTemplate components from one Salesforce org to another, you must use change sets.’

Have you seen a successful deployment for this meta data?

Best,
-Sheida

**Reply by scolladon** [reply]
> Have you seen a successful deployment for this meta data?

Hi @sheidasf, I had not this chance yet...

I looked at [metadata coverage](https://developer.salesforce.com/docs/metadata-coverage/61) and could not find information about the [GenAIPromptTemplate](https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_genaiprompttemplate.htm)...
And a quick search on google gave nothing as well...

My bet is you should create an issue to @salesforce/cli repository to ask them to include this metadata. Maybe they will be able to do it, or they will be able to route it to the right team.
